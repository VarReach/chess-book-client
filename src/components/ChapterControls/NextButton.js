import React, { Component } from 'react';
import UserContext from '../../contexts/UserContext';
import { Link } from 'react-router-dom';

export default class NextButton extends Component {
  static contextType = UserContext;

  render() {
    return (
      <Link
          className="article__chapter-next-button"
          to={"/"}
          onClick={(e) => {
              this.props.handleOnClick(e, this.props.bookId, this.props.chapterIndex + 1);
              this.context.completeChapter(this.props.chapterId);
            }
          }
      >
        <i className='fas fa-angle-right'/>
      </Link>
    )
  }
}