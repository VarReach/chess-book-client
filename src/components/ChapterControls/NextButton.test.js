import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NextButton from './NextButton';

describe('NextButton component', () => {
  it('renders the button', () => {
    const wrapper = shallow(<NextButton/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});