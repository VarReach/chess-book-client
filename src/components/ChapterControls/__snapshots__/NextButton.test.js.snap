// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextButton component renders the button 1`] = `
<Link
  className="article__chapter-next-button"
  onClick={[Function]}
  replace={false}
  to="/"
>
  <i
    className="fas fa-angle-right"
  />
</Link>
`;
